### ISSUE TITLE
Binding interface does not support excess properties

### ISSUE THREAD

**Issue by oviecodes** [asyncapi/parser-js#735]
#### Describe the bug
The OldServer, ServerV2, and Server classes do not support excess properties on the Binding interface.

#### How to Reproduce

using "@asyncapi/parser": "^2.0.0-next-major.15"

```ts
import { OldAsyncAPIDocument, OldServer } from '@asyncapi/parser'

//inside an exported class
private _AsyncAPIServer: OldServer

get AsyncAPIServer(): OldServer {
    return this._AsyncAPIServer
}

//in another file with a class that extends the class above

const mqttServerBinding = this.AsyncAPIServer.binding('mqtt')
const mqtt5ServerBinding = this.AsyncAPIServer.binding('mqtt5')

const serverBinding = protocolVersion === 5 ? mqtt5ServerBinding : mqttServerBinding

mqtt.connect({
      ...
      clientId: serverBinding?.clientId ?? auth?.clientId,
      clean: serverBinding?.cleanSession,
      ...
})
```

An error is thrown that `clientId` does not exist on type `Binding`.

![glee_error](https://user-images.githubusercontent.com/61796959/227504358-cc325f08-9ada-4245-9cba-070c0cdb2dfa.png)

The Binding interface 
![glee_binding](https://user-images.githubusercontent.com/61796959/227504566-920688a6-d674-490a-8c54-b578e5930a2b.png)

#### Expected behavior
The Binding interface should support excess properties possibly by using `[propName: string]: any;`

**Comment by magicmatatjahu** [comment]
@oviecodes Do you wanna add PR?

**Comment by oviecodes** [comment]
Alright @magicmatatjahu , I'll do so

**Comment by oviecodes** [comment]
@magicmatatjahu I also noticed that the `AsyncApiDocumentV2` interface doesn't have the `channelNames()` property. Was this done on purpose?

**Comment by jonaslagoni** [comment]
Cc @smoya can you remember whether this was addressed in one of the recent changes?

**Comment by smoya** [comment]
> Cc @smoya can you remember whether this was addressed in one of the recent changes?

No, you will still need to do something like `(mqttServerBinding as any).clientId` to fetch the value. 

@oviecodes are you ok creating the PR for this?

**Comment by oviecodes** [comment]
yeah @smoya , just waiting for @magicmatatjahu to respond then it's a go

**Comment by magicmatatjahu** [comment]
Go with this.

**Comment by oviecodes** [comment]
Alright, I'll create a PR

**Comment by tzh476** [comment]
Hi, I would like to take this issue as part of the AsyncAPI microgrant program. I already implemented and verified the agreed fix: adding the `[propName: string]: any;` index signature to both the v2 and v3 `Binding` spec-type interfaces (per the maintainer-approved approach), plus a regression test. Verified: `tsc --noEmit` clean, v2 bindings 9/9, v3 bindings 8/8. Could I be assigned to submit the PR? Thanks.
